statsd/buffered_metric_context.go

```diff
@@ -57,6 +57,15 @@ func (bc *bufferedMetricContexts) flush(metrics []metric) []metric {
 func (bc *bufferedMetricContexts) sample(name string, value float64, tags []string, rate float64) error {
 	keepingSample := shouldSample(rate, bc.random, &bc.randomLock)
 
+	// If we don't keep the sample, return early, if we do store *first* observed sampling rate in the metric.
```

Comment:
I'm having a hard time understanding this comment, here's a suggestion
```suggestion
	// If we don't keep the sample, return early. If we do keep the sample
	// we end up storing the *first* observed sampling rate in the metric.
```